tests/chainer_tests/test_sequential.py

```diff
@@ -607,4 +601,69 @@ def test_empty_sequential(self):
             seq(x)
 
 
+class SequentialInsert(unittest.TestCase):
```

Comment:
How about adding tests inserting `Link`s (similar to the bug report https://github.com/chainer/chainer/issues/6325#issue-412548070)? The code paths are different for functions and links.

Reply:
Hmm, code paths are not different after the fix. What do you mean?
The test complexity would be greatly increase which I don't think worthwhile.

Reply:
The paths are still different if the index is not out of bounds. How about restoring https://github.com/chainer/chainer/pull/6374/files#diff-a3da12ecfbd5149195d80196a0b251c1L424 ?

Reply:
`TestSequentialInsert` should test inserting `Link`s, while `Sequential.insert(_, link)` is called elsewhere in the file.  Having both `TestSequential.test_insert` and `TestSequentialInsert` is confusing.

Reply:
I believe that's a perfectly valid point. My point was that we should still test inserting links, not necessarily with all combinations of invalid indices. We could do that by more or less moving the removed method to this class.

Reply:
It would be insufficient to literally restoring the single test.

Reply:
Fixed in 5030fb7c304cff0813d391596d1fa0114e436acd.